### DISCUSSION TITLE
How to increase time-out when uploading to a storage bucket

### DISCUSSION THREAD

**Discussion by lablack576** [supabase/supabase#23218] [Questions] [answered]
Hey, I've been using Supabase with the js sdk to upload a video into my storage bucket but since the video is a big file and my Internet is slow it's taking sometimes to upload then returning an error saying it reached the time-out.

API Call :

   const formData = new FormData();
      formData.set(file, data.file);
      const res = await fetch(/api/sending, {
        method: "POST",
        body: formData,
      });

API Route :

export async function POST(request: NextRequest) {
await supabase.storage
        .from("videos")
        .upload(video_1, file);
    return NextResponse.json({
        success: true,
        message: "Processing completed successfully.",
      });
}

Any idea how to increase it, it's currently 5 minutes !

**Comment by GaryAustin1** [comment, marked as answer]
What is the size of your file?  This recommends using resumable uploads for large files (>5MB).
https://supabase.com/docs/guides/storage/uploads/standard-uploads
https://supabase.com/docs/guides/storage/uploads/resumable-uploads

**Reply by lablack576** [reply]
I think you're right the size of the file is larget than 5MB.
